### PR TITLE
Chatbot autoscroll

### LINKED ISSUE #8253: Add auto-scroll down to the bottom of the Gradio chat box whenever there's a new LLM response or show down arrow button
- Yes, I have searched to see if a similar issue already exists.

While using the Gradio chat interface I noticed that when I scroll up to see previous LLM responses, and then submit a new question, the latest LLM response is not shown. The chat box does not automatically scroll down to the bottom to show the latest response. It just stays there.

What would be nice enhancement to see (and I think it would be helpful for many users) is that when there's a new answer/response, it should scroll down automatically and show the latest response (bubble). Alternatively, if not auto-scroll, then it should display a down arrow at the bottom center of the chatbox textarea to let the user know there's a new response which when clicked should scroll down to the latest response bubble (similar to ChatGPT interface).

Example Solution (down arrow button indicating there's new response):
image0

### PR DESCRIPTION
**Pull request by whitphx**
## Description

Resolves #8253

* Add `autoscroll` parameter to `gr.Chatbot` and `gr.ChatInterface` to turn on/off the autoscroll feature. It's enabled by default.
* There has already been the auto-scroll impl on the Chatbot component but looked like it didn't work. This PR fixes it.
* As suggested in #8253 , a "scroll-down" button is added.

`multimodal=False`:

https://github.com/user-attachments/assets/04acf253-4394-45a6-9f0d-6b90a60c4103

`multimodal=True` (multimodal is [broken now](https://huggingface.slack.com/archives/C07N8RBSR39/p1728288842304229) so I tested this with local patches)

https://github.com/user-attachments/assets/55b266c8-7385-401e-a43d-4ff1e1c266af

### CODE REVIEW

File: js/chatbot/shared/ScrollDown.svelte [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<svg
```
Comment (whitphx) [inline js/chatbot/shared/ScrollDown.svelte:1]: I just made this SVG with LLM for this component, but plz tell me if there is already a usable resource for this purpose.
Also, should this be put in the `@gradio/icons` package?

**Comment by whitphx** [comment]
Auto-scroll doesn't work sometimes with media files. Will investigate it.

**Comment by whitphx** [comment]
Fixed!

File: js/chatbot/shared/ScrollDown.svelte (hunk 1 above)
Comment (dawoodkhan82) [reply js/chatbot/shared/ScrollDown.svelte:1]: Yes, can you add it to the icons package.

Comment (dawoodkhan82) [reply js/chatbot/shared/ScrollDown.svelte:1]: Fixed

**Review by dawoodkhan82 — APPROVED** [review]

**Comment by Matteo-Bassani** [comment]
If you have a Chatbot inside a gr.Tab the autoscroll doesn't work at all. 

import gradio as gr
import random
import time

with gr.Blocks() as demo:
    with gr.Tab("Chatbot"):
        chatbot = gr.Chatbot(type="messages")
        msg = gr.Textbox()
        clear = gr.ClearButton([msg, chatbot])

        def respond(message, chat_history):
            bot_message = random.choice(["How are you?", "Today is a great day", "I'm very hungry"])
            chat_history.append({"role": "user", "content": message})
            chat_history.append({"role": "assistant", "content": bot_message})
            time.sleep(2)
            return "", chat_history

    with gr.Tab("Another Tab"):
        gr.Textbox()

    msg.submit(respond, [msg, chatbot], [msg, chatbot])

demo.launch()

**Pull request merged**